dnsprovider/google.go

```diff
@@ -212,24 +185,25 @@ func (p *GoogleProvider) DeleteRecords(zone string, records []endpoint.Endpoint)
 	return nil
 }
 
-// ApplyChanges applies a given set of changes in a given zone.
-func (p *GoogleProvider) ApplyChanges(zone string, changes *plan.Changes) error {
-	err := p.CreateRecords(zone, changes.Create)
-	if err != nil {
-		return err
-	}
+// newRecords returns a collection of RecordSets based on the given endpoints.
+func newRecords(endpoints []endpoint.Endpoint) []*dns.ResourceRecordSet {
+	records := make([]*dns.ResourceRecordSet, 0, len(endpoints))
```

Comment:
just out of interest, append() in some cases can be faster, afaik in this case, i.e. with preallocated correct capacity, append could be interchangeably used with direct slice element index access